internal/stacktrace/stacktrace.go

```diff
@@ -0,0 +1,195 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+//go:generate msgp -o=stacktrace_msgp.go -tests=false
+
+package stacktrace
+
+import (
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-secure-stdlib/parseutil"
+)
+
+var enabled = true
+var defaultTopFrameDepth = 8
+var defaultMaxDepth = 32
+
+const defaultCallerSkip = 3
+const stackTraceDepthEnvVar = "DD_APPSEC_MAX_STACK_TRACE_DEPTH"
+const stackTraceDisabledEnvVar = "DD_APPSEC_STACK_TRACE_ENABLE"
+
+func init() {
+	if env := os.Getenv(stackTraceDepthEnvVar); env != "" {
+		if depth, err := parseutil.SafeParseInt(env); err == nil {
+			defaultMaxDepth = depth
+		}
+	}
+
+	defaultTopFrameDepth = defaultMaxDepth / 4
+
+	if env := os.Getenv(stackTraceDisabledEnvVar); env != "" {
+		if e, err := strconv.ParseBool(env); err == nil {
+			enabled = e
+		}
+	}
+}
+
+// StackTrace is intended to be sent over the span tag `_dd.stack`, the first frame is the current frame
+type StackTrace []StackFrame
+
+// StackFrame represents a single frame in the stack trace
+type StackFrame struct {
```

Comment:
Optional -- the package is called `stacktrace`, so I think you can reduce stutter so that it's:

```suggestion
// Trace is intended to be sent over the span tag `_dd.stack`, the first frame is the current frame
type Trace []Frame

// Frame represents a single frame in the stack trace
type Frame struct {
```

Reply:
That's the first thing I did actually. But a `Trace` is too vague in the tracer, if we end up in cases were the package name disappear for some reasons (aliases all kind) This would be too confusing So I keep it that way. But for example, `Event` is less confusing so I did not double down on the stacktrace name.

Reply:
Since this is marked optional I will resolve the thread